### PR TITLE
Merge policy tests

### PR DESCRIPTION
**Pull request by OHerasym**
### PR provides following changes:
- Fix policy tests

Correct tests according to code changes and make tests buildable

> Tests builds on LINUX & WINDOWS & QT platforms

32 tests disabled : _(thread qt dcheck fails)_
- usage_statistics_test.cc:136:    DISABLED_AppStopwatchStartMethod_CallONCE_StatisticsManagerAddMethodCalledONCE) {
- usage_statistics_test.cc:150:     DISABLED_AppStopwatchStartMethod_Call_StatisticsManagerAddMethodCALLED) {
- usage_statistics_test.cc:166:     DISABLED_AppStopwatchSwitchMethod_Call_StatisticsManagerAddMethodCALLED) {
- usage_statistics_test.cc:182:    DISABLED_AppStopwatchStartMethod_CallAnd1SecSleepAfter_StatisticsManagerAddMethodCalledWith1SecTimespan) {
- usage_statistics_test.cc:197:    DISABLED_AppStopwatchSwitchMethod_CallAnd1SecSleepAfter_StatisticsManagerAddMethodCalledWith1SecTimespan) {
- shared_library_test.cc:51:    DISABLED_FullTest_OpenLibrarySetSymbolCloseLibrary_ExpectActsWithoutErrors) {
- policy_manager_impl_test.cc:421:TEST_F(PolicyManagerImplTest, DISABLED_GetNotificationsNumber) {
- policy_manager_impl_test.cc:430:TEST_F(PolicyManagerImplTest2, DISABLED_GetNotificationsNumberAfterPTUpdate) {
- policy_manager_impl_test.cc:468:       DISABLED_IsAppRevoked_SetRevokedAppID_ExpectAppRevoked) {
- policy_manager_impl_test.cc:485:TEST_F(PolicyManagerImplTest, DISABLED_IncrementGlobalCounter) {
- policy_manager_impl_test.cc:491:TEST_F(PolicyManagerImplTest, DISABLED_IncrementAppCounter) {
- policy_manager_impl_test.cc:498:TEST_F(PolicyManagerImplTest, DISABLED_SetAppInfo) {
- policy_manager_impl_test.cc:505:TEST_F(PolicyManagerImplTest, DISABLED_AddAppStopwatch) {
- policy_manager_impl_test.cc:512:TEST_F(PolicyManagerImplTest, DISABLED_ResetPT) {
- policy_manager_impl_test.cc:524:TEST_F(PolicyManagerImplTest, DISABLED_LoadPT_SetPT_PTIsLoaded) {
- policy_manager_impl_test.cc:555:       DISABLED_LoadPT_SetInvalidUpdatePT_PTIsNotLoaded) {
- policy_manager_impl_test.cc:581:       DISABLED_KmsChanged_SetExceededKms_ExpectCorrectSchedule) {
- policy_manager_impl_test.cc:597:    DISABLED_AddApplication_AddNewApplicationFromDeviceWithoutConsent_ExpectUpdateRequired) {
- policy_manager_impl_test.cc:607:    DISABLED_AddApplication_AddExistingApplicationFromDeviceWithoutConsent_ExpectNoUpdateRequired) {
- policy_manager_impl_test.cc:621:       DISABLED_PTUpdatedAt_DaysNotExceedLimit_ExpectNoUpdateRequired) {
- policy_manager_impl_test.cc:642:TEST_F(PolicyManagerImplTest2, DISABLED_ForcePTExchange_ExpectUpdateNeeded) {
- policy_manager_impl_test.cc:652:TEST_F(PolicyManagerImplTest2, DISABLED_OnSystemReady) {
- policy_manager_impl_test.cc:660:TEST_F(PolicyManagerImplTest2, DISABLED_ResetRetrySequence) {
- policy_manager_impl_test.cc:669:TEST_F(PolicyManagerImplTest2, DISABLED_NextRetryTimeout_ExpectTimeoutsFromPT) {
- policy_manager_impl_test.cc:694:TEST_F(PolicyManagerImplTest2, DISABLED_TimeOutExchange) {
- policy_manager_impl_test.cc:701:TEST_F(PolicyManagerImplTest2, DISABLED_GetPolicyTableStatus_ExpectUpToDate) {
- policy_manager_impl_test.cc:709:       DISABLED_RetrySequenceDelaysSeconds_Expect_CorrectValues) {
- policy_manager_impl_test.cc:730:       DISABLED_OnExceededTimeout_GetPolicyTableStatus_ExpectUpdateNeeded) {
- policy_manager_impl_test.cc:740:       DISABLED_GetInitialAppData_ExpectReceivedConsentCorrect) {
- policy_manager_impl_test.cc:783:    DISABLED_CanAppKeepContext_SetPoliciesForAppUpdated_ExpectAppCanKeepContext) {
- policy_manager_impl_test.cc:795:    DISABLED_CanAppStealFocus_SetPoliciesForAppUpdated_ExpectAppCanStealFocus) {
- policy_manager_impl_test.cc:804:TEST_F(PolicyManagerImplTest2, DISABLED_GetCurrentDeviceId) {
- policy_manager_impl_test.cc:812:       DISABLED_GetVehicleInfo_SetVehicleInfo_ExpectReceivedInfoCorrect) {
- counter_test.cc:141:    DISABLED_AppStopwatchStartMethod_CallONCE_StatisticsManagerAddMethodCalledONCE) {
- counter_test.cc:156:     DISABLED_AppStopwatchSwitchMethod_Call_StatisticsManagerAddMethodCalled) {
- counter_test.cc:172:    DISABLED_AppStopwatchSwitchMethod_CallAnd1SecSleepAfter_StatisticsManagerAddMethodCalledWith1SecTimespan) {
- update_status_manager_test.cc:65:    DISABLED_StringifiedUpdateStatus_SetStatuses_ExpectCorrectStringifiedStatuses) {
- update_status_manager_test.cc:79:       DISABLED_OnAppSearchStartedCompleted_ExpectAppSearchCorrectStatus) {
- sql_pt_representation_test.cc:404:       DISABLED_OpenAttemptTimeOut_ExpectCorrectNumber) {
- sql_pt_representation_test.cc:1029:     DISABLED_Init_TryInitNotExistingDataBase_ExpectResultFail) {
- sql_pt_representation_test.cc:1042:     DISABLED_Close_InitNewDataBaseThenClose_ExpectResultSuccess) {
- sql_pt_representation_test.cc:1464:TEST(SQLPTRepresentationTest3, DISABLED_RemoveDB_RemoveDB_ExpectFileDeleted) {
- sql_pt_representation_test.cc:1481:       DISABLED_GenerateSnapshot_SetPolicyTable_SnapshotIsPresent) {
- sql_pt_representation_test.cc:1522:       DISABLED_Save_SetPolicyTableThenSave_ExpectSavedToPT) {

This PR contains changes and review notes fixes from [previous PR](https://github.com/OHerasym/sdl_core/pull/4).

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @AByzhynar, @dev-gh

File: src/components/policy/test/counter_test.cc [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 #include "gmock/gmock.h"
 
-#include "mock_statistics_manager.h"
+#include "include/test/policy/mock_statistics_manager.h"
```
Comment (AByzhynar) [inline src/components/policy/test/counter_test.cc]: @OHerasym Please edit header path . Should be used the following template : `<directory/file.h>`. Path to `directory` should be specified in CMakeList

File: src/components/policy/test/generated_code_test.cc [hunk 2]
```diff
@@ -48,21 +49,29 @@ namespace policy {
...
-  ASSERT_TRUE(reader.parse(json_file, valid_table));
-  Table table(&valid_table);
+  std::stringstream buffer;
+  buffer << json_file.rdbuf();
+  std::string str(buffer.str());
+  utils::json::JsonValue::ParseResult result =
+      utils::json::JsonValue::Parse(str);
+  utils::json::JsonValue valid_table = result.first;
+  utils::json::JsonValueRef json_value_ref = valid_table;
+  Table table(json_value_ref);
```
Comment (AByzhynar) [inline src/components/policy/test/generated_code_test.cc:59]: @OHerasym In bot tests you use absolutely the same code. Please move it out to separate function and use just function call.

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 3]
```diff
@@ -37,28 +37,29 @@
 
 #include "json/reader.h"
 #include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "mock_policy_listener.h"
+#include "mock_pt_representation.h"
+#include "policy/test/include/mock_cache_manager.h"
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym Please edit header path . Should be used the following template : `<directory/file.h>`. Path to `directory` should be specified in CMakeList

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 4]
```diff
@@ -104,7 +106,7 @@ class PolicyManagerImplTest : public ::testing::Test {
   NiceMock<MockPolicyListener> listener;
 
   void SetUp() OVERRIDE {
-    manager = new PolicyManagerImpl();
+    manager = new PolicyManagerImpl("testing", 1, 10000);
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym Magic numbers? Why 10000 and not 20000? May be it's worth to use some named constants?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 5]
```diff
@@ -150,12 +152,11 @@ class PolicyManagerImplTest2 : public ::testing::Test {
   const std::string kAppStorageFolder = "storage1";
 
   void SetUp() OVERRIDE {
+    DefaultValue<const std::string&>::Set(kAppStorageFolder);
     file_system::CreateDirectory("storage1");
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym Please use `kAppStorageFolder` as function parameter

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 6]
```diff
@@ -188,10 +189,8 @@ class PolicyManagerImplTest2 : public ::testing::Test {
     return root;
   }
 
-  void CreateLocalPT(const std::string& file_name) {
-    file_system::remove_directory_content("storage1");
-    ON_CALL(policy_settings_, app_storage_folder())
-        .WillByDefault(ReturnRef(kAppStorageFolder));
+  void CreateLocalPT(std::string file_name) {
+    file_system::RemoveDirectoryContent("storage1");
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym Please use `kAppStorageFolder` as function parameter

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 7]
```diff
@@ -418,16 +418,17 @@ TEST_F(PolicyManagerImplTest, GetNotificationsNumber) {
...
   update.SetPolicyTableType(rpc::policy_table_interface_base::PT_UPDATE);
   // Act
-  const std::string json = table.toStyledString();
+  const std::string json = table.ToJson(true);
   ::policy::BinaryMessage msg(json.begin(), json.end());
   EXPECT_CALL(listener, OnUpdateStatusChanged(_));
   EXPECT_TRUE(manager->LoadPT("file_pt_update.json", msg));
 
-  std::string priority = "emergency";
+  std::string priority = "EMERGENCY";
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc:442]: @OHerasym const?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 8]
```diff
@@ -510,8 +511,8 @@ TEST_F(PolicyManagerImplTest, ResetPT) {
 
 TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
   // Arrange
-  Json::Value table = CreatePTforLoad();
-  policy_table::Table update(&table);
+  utils::json::JsonValue table = CreatePTforLoad();
+  policy_table::Table update(table);
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc:527]: @OHerasym May be use : `policy_table::Table update(CreatePTforLoad());` ?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 9]
```diff
@@ -540,16 +541,16 @@ TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
...
-  policy_table::Table update(&table);
+  policy_table::Table update(table);
   update.SetPolicyTableType(rpc::policy_table_interface_base::PT_UPDATE);
 
   // Assert update is invalid
   ASSERT_FALSE(IsValid(update));
 
   // Act
-  std::string json = table.toStyledString();
+  std::string json = table.ToJson(true);
```
Comment (AByzhynar) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym Does not need to be const?

File: src/components/policy/test/sql_pt_representation_test.cc [hunk 10]
```diff
@@ -336,7 +351,7 @@ std::auto_ptr<policy_handler_test::MockPolicySettings>
 class SQLPTRepresentationTest2 : public ::testing::Test {
  protected:
   SQLPTRepresentationTest2()
-      : kAppStorageFolder("storage123")
+      : kAppStorageFolder("storage1")
```
Comment (AByzhynar) [inline src/components/policy/test/sql_pt_representation_test.cc:373]: @OHerasym Use named const like above

File: src/components/policy/test/sql_pt_representation_test.cc [hunk 11]
```diff
@@ -65,15 +66,24 @@ using testing::ReturnRef;
...
 
 namespace test {
 namespace components {
 namespace policy {
 
+namespace {
+const std::string kStorageFolder = ".";
+const int kAttemptsToOpenPolicyDb = 5;
+const int kOpenAttemptTimeoutMs = 7000;
+}
```
Comment (AByzhynar) [inline src/components/policy/test/sql_pt_representation_test.cc]: @OHerasym Should be finished with `\\ namespace`

File: src/components/policy/test/sql_pt_representation_test.cc (hunk 10 above)
Comment (OHerasym) [inline src/components/policy/test/sql_pt_representation_test.cc:373]: @AByzhynar , It is const initialization

File: src/components/policy/test/sql_pt_representation_test.cc (hunk 11 above)
Comment (OHerasym) [inline src/components/policy/test/sql_pt_representation_test.cc]: @AByzhynar ,    e9d0b82 , Done

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 6 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AByzhynar , Done

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 5 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AByzhynar , Done

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 4 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AByzhynar , Done

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 3 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AByzhynar , Done

File: src/components/policy/test/counter_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/policy/test/counter_test.cc]: @AByzhynar , DOne

**Comment by OHerasym** [comment]
@AByzhynar , other comments will be fixed after UT refactoring

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 7 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc:442]: @AByzhynar , no, priority will be changed during test executing

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 8 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc:527]: @AByzhynar , I'm not agree, your version is harder to understand

File: src/components/policy/test/generated_code_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/policy/test/generated_code_test.cc:59]: @AByzhynar , will be done after UT refactoring

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 9 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AByzhynar, Done

**Comment by AByzhynar** [comment]
Reviewed

**Pull request merged**